styleguide/js/zoolander.js

```diff
@@ -126,6 +128,114 @@ jQuery(document).ready(function($) {
   };
 }(jQuery)); //to protect and scope the JQuery alias = $
 
+// Filter bar plugin
+(function($){
+  'use strict';
+  $.fn.rsFilterBar = function(){
+    // Variable declorations
+    var $filterBar = $(this);
+    var $filterForm = $filterBar.find('.rsFilter-form');
+    var $stickyItemTop = $filterForm.offset().top;
+    var $mainNavHeight = parseInt($('.navbar-fixed-top').outerHeight(), 10);
+    var $barPadding = parseInt($filterBar.css('padding-left'), 10) +
+                      parseInt($filterBar.css('padding-right'), 10);
+    var $filterHamburger = $filterBar.find('.rsFilter-hamburger');
+    var $rsWindow = $(window);
+    var isFixed = false;
+    // var inBounds = false;
+
+    // Callback to get new form width on resizing/scrolling
+    function getWidth(bar, padding) {
+      var width = parseInt(bar.outerWidth(true), 10) - padding;
+      return width;
+    }
+
+    // Callback to check if in bounds of side menu
+    function inBounds(wt) {
+      return $stickyItemTop - $mainNavHeight < wt &&
+        $filterBar.outerHeight() + $filterBar.offset().top +
+        $mainNavHeight - ($filterForm.outerHeight() * 3) >
+        (wt + $filterForm.outerHeight())
```

Comment:
So I know I previously recommended just flat out returning this, but now that I see it, I might recommend tweaking it a little bit more, as it's a bit confusing as to what's going on here. This is a rather long condition and doesn't really tell us a whole lot of what is actually going on.  Perhaps something like this would be better?
```
    function inBounds(wt) {
      var filterFormHeight = $filterForm.outerHeight();
      var windowTopInBounds = $stickyItemTop - $mainNavHeight < wt;
      var filterBottom = $filterBar.outerHeight() + $filterBar.offset().top;
      var formBottom = wt + filterFormHeight;
      var filterInBounds = filterBottom + $mainNavHeight - filterFormHeight * 3 > formBottom;
      return windowTopInBounds && filterInBounds;
    }
```
Something like this feels a little bit more expressive in my opinion, and won't have us scratching our heads 6 months down the road trying to figure out what was going on here. 

Additionally, this is a prime example of why it would be good to have JS unit testing setup in zoolander.  Whenever testing does get added, I'd recommend this function be one of the first things to get tested.